mgr/src/cmds/pools.cr

```diff
@@ -1,79 +1,345 @@
+    args.pool_args.volfiles_separator = sep
+  end
+end
+
+def pool_detail(pool, args)
+  status = args.pool_args.status
+  health = pool.state == "Started" && status ? "#{pool.state} (#{pool.metrics.health})" : pool.state
+
+  volfile_servers = [] of String
+  pool.distribute_groups.each do |dist_grp|
+    dist_grp.storage_units.each do |storage_unit|
+      volfile_servers << "#{storage_unit.node.name}:#{storage_unit.port}"
+    end
+  end
+
+  puts "Name                    : #{pool.name}"
+  puts "Type                    : #{pool.type}"
+  puts "ID                      : #{pool.id}"
+  puts "State                   : #{health}"
+  puts "Size                    : #{(pool.metrics.size_used_bytes + pool.metrics.size_free_bytes).humanize_bytes}"
+  puts "Inodes                  : #{(pool.metrics.inodes_used_count + pool.metrics.inodes_free_count).humanize}"
+  puts "Utilization             : #{pool.metrics.size_used_bytes.humanize_bytes}/#{(pool.metrics.size_used_bytes + pool.metrics.size_free_bytes).humanize_bytes}" if status
+  puts "Utilization (Inodes)    : #{pool.metrics.inodes_used_count.humanize}/#{(pool.metrics.inodes_used_count + pool.metrics.inodes_free_count).humanize}" if status
+  puts "Volfile Servers         : #{volfile_servers.join(args.pool_args.volfiles_separator)}"
+  puts "Options                 :#{pool.options.size > 0 ? "" : " -"}"
+
+  pool.options.each do |k, v|
+    printf("    %20s: %s\n", k, v)
+  end
+  puts "Number of storage units : #{pool.distribute_groups.size * pool.distribute_groups[0].storage_units.size}"
+  pool.distribute_groups.each_with_index do |dist_grp, grp_idx|
+    printf("Distribute group %-2s     :%s\n", grp_idx + 1, status ? " Health: #{dist_grp.metrics.health}" : "")
+    dist_grp.storage_units.each_with_index do |storage_unit, idx|
+      printf(
+        "    Storage unit %-3s    : %s:%s (Port: %s%s)",
+        idx + 1,
+        storage_unit.node.name,
+        storage_unit.path,
+        storage_unit.port,
+        status ? ", Health: #{storage_unit.metrics.health}" : ""
+      )
+      puts
+    end
+  end
+
+  puts
 end
 
 handler "pool.list" do |args|
-  api_call(args, "Failed to get the list of Pools") do |client|
-    pools = client.list_pools
+  args.pool_args.name = args.pos_args.size < 1 ? "" : args.pos_args[0]
+
+  api_call(args, "Failed to get the list of pools") do |client|
+    if args.pool_args.name == ""
+      pools = client.list_pools(state: args.pool_args.status)
+    else
+      pools = [client.pool(args.pool_args.name).get(state: args.pool_args.status)]
+    end
 
     handle_json_output(pools, args)
 
-    puts "No pools. Run `kadalu pool create <name>` to create a Pool." if pools.size == 0
+    if pools.size == 0
+      puts "No pools available in the Cluster.\nRun `kadalu pool create \"<pool>\" ...` to create a pool."
+    end
+
+    if args.pool_args.detail
+      pools.each do |pool|
+        pool_detail(pool, args)
+      end
+
+      next
```

Comment:
Usage of next here?

Reply:
same as `return` in function. Use `next` in blocks. This file is renamed from volumes.cr to pools.cr and then Volume changed to Pool. Diff is showing it as full new implementation :(